src/main/java/org/javarosa/core/model/utils/DateUtils.java

```diff
@@ -89,20 +91,25 @@ public DateFields () {
             this.second = second;
             this.secTicks = secTicks;
             this.dow = dow;
+            this.week = week;
         }
 
         public static DateFields of(int year, int month, int day, int hour, int minute, int second, int secTicks) {
             // The official API returns an ISO 8601 day of week
             // with a range of values from 1 for Monday to 7 for Sunday].
-            // TODO migrate dow field to a DayOfWeek type to avoid any possible interpretation errors
-            int iso8601Dow = new LocalDateTime(year, month, day, hour, minute, second, secTicks).getDayOfWeek();
+            // TODO migrate dow field to a DayOfWeek type to avoid any possible
+            // interpretation errors
+            LocalDateTime ldt = new LocalDateTime(year, month, day, hour, minute, second, secTicks);
+            int iso8601Dow = ldt.getDayOfWeek();
             int dow = iso8601Dow == 7 ? 0 : iso8601Dow;
-            return new DateFields(year, month, day, hour, minute, second, secTicks, dow);
+            int week = ldt.getWeekOfWeekyear();
+            return new DateFields(year, month, day, hour, minute, second, secTicks, dow, week);
         }
 
-        public boolean check () {
-            return (inRange(month, 1, 12) && inRange(day, 1, daysInMonth(month - MONTH_OFFSET, year)) &&
-                    inRange(hour, 0, 23) && inRange(minute, 0, 59) && inRange(second, 0, 59) && inRange(secTicks, 0, 999));
+        public boolean check() {
+            return (inRange(month, 1, 12) && inRange(day, 1, daysInMonth(month - MONTH_OFFSET, year))
+                    && inRange(hour, 0, 23) && inRange(minute, 0, 59) && inRange(second, 0, 59)
+                    && inRange(secTicks, 0, 999) && inRange(week, 1, 52));
```

Comment:
The week range should be 1-53. See https://en.wikipedia.org/wiki/ISO_week_date.

Reply:
fixed